### PR TITLE
Move update of user profile to users controller

### PR DESCRIPTION
**Pull request by letiesperon**
#### Description:

This PR refactors the user routes to make them singular for update and show. 

Previously they acted on the collection 
* `PUT /api/v1/users` to update your profile. It now becomes `PUT /api/v1/user`
* `GET /api/v1/users` to see your profile. It now becomes `GET /api/v1/user`

Also, it changes the routes so that the update to the user profile goes to users controller instead of registration controller (to have more explicit control over it.)

It also deletes unnecessary registrations routes like destroying of a user account (might be dangerous in some systems, you not always want to permit that action). 

It also allows the frontend to change the user password, requiring a password check when doing so. 

---

@loopstudio/ruby-devs

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -3,13 +3,29 @@ module V1
...
+      end
+
       private
 
       def user
         @user ||= current_user
       end
+
+      def password_check
+        return if !user_params[:password] || user.valid_password?(params[:password_check])
```
Comment (lucaleivaloop) [inline app/controllers/api/v1/users_controller.rb:21]: You are requiring the old password to update the password, right? If that is the case maybe you can extract the part after the if to make it more readable. What do you think?

Comment (letiesperon) [reply app/controllers/api/v1/users_controller.rb:21]: I don't think I understood

Comment (lucaleivaloop) [reply app/controllers/api/v1/users_controller.rb:21]: Maybe you can do something of the sorts of:
```suggestion
        return if updating_password? && has_current_password?
```
Or rename `password_check` to something like `check_for_password_update`.

Comment (letiesperon) [reply app/controllers/api/v1/users_controller.rb:21]: password_check  is a param that comes in the body and should equal the user password, I don't think I fully understand your suggestion

**Pull request merged**